mmv1/products/firestore/BackupSchedule.yaml

```diff
@@ -0,0 +1,114 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'BackupSchedule'
+base_url: projects/{{project}}/databases/{{database}}/backupSchedules
+self_link: 'projects/{{project}}/databases/{{database}}/backupSchedules/{{name}}'
+update_verb: :PATCH
+update_mask: true
+description: |
+  A backup schedule for a Cloud Firestore Database.
+  This resource is owned by the database it is backing up, and is deleted along with the database.
+  The actual backups are not though.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/firestore/docs/backups'
+  api: 'https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases.backupSchedules'
+skip_sweeper: true
+id_format: 'projects/{{project}}/databases/{{database}}/backupSchedules/{{name}}'
+import_format:
+  - 'projects/{{project}}/databases/{{database}}/backupSchedules/{{name}}'
+  - '{{project}}/{{database}}/{{name}}'
+  - '{{database}}/{{name}}'
+docs: !ruby/object:Provider::Terraform::Docs
+  warning: |
+    This resource creates a Firestore Backup Schedule on a project that already has
+    a Firestore database. If you haven't already created it, you may
+    create a `google_firestore_database` resource with `type` set to
+    `"FIRESTORE_NATIVE"` and `location_id` set to your chosen location.
+    If you wish to use App Engine, you may instead create a
+    `google_app_engine_application` resource with `database_type` set to
+    `"CLOUD_FIRESTORE"`. Your Firestore location will be the same as
+    the App Engine location specified.
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'firestore_backup_schedule_daily'
+    primary_resource_id:
+      'daily-backup'
+    test_env_vars:
+      project_id: :FIRESTORE_PROJECT_NAME
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'firestore_backup_schedule_weekly'
+    primary_resource_id:
+      'weekly-backup'
+    test_env_vars:
+      project_id: :FIRESTORE_PROJECT_NAME
+parameters:
+  - !ruby/object:Api::Type::String
+    name: database
+    description: |
+      The Firestore database id. Defaults to `"(default)"`.
+    default_value: '(default)'
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: name
+    output: true
+    description: |
+      A server defined name for this index. Format:
+      `projects/{{project}}/databases/{{database}}/backupSchedules/{{backupSchedule}}
+    immutable: true
+    custom_flatten: templates/terraform/custom_flatten/name_from_self_link.erb
+  - !ruby/object:Api::Type::String
+    name: retention
+    required: true
```

Comment:
Is `retention` a required field? What happens if it is not included in a request to the API? The API [documentation doesn't note that the field is required](https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases.backupSchedules#BackupSchedule.FIELDS.retention), but that might just be an inaccuracy in the docs.

Reply:
When trying to create it without it leads to the following error:

```
{
  "error": {
    "code": 400,
    "message": "\u003ceye3 title='/FirestoreAdmin.CreateBackupSchedule, INVALID_ARGUMENT'/\u003e APPLICATION_ERROR;google.firestore.admin.v1/FirestoreAdmin.CreateBackupSchedule;Retention must be specified in the BackupSchedule.;AppErrorCode=3;StartTimeMs=1696353643353;unknown;ResFormat=uncompressed;ServerTimeSec=0.001389596;LogBytes=256;Non-FailFast;EffSecLevel=none;ReqFormat=uncompressed;ReqID=1e6fb03d0ec10188;GlobalID=0;Server=[2002:a05:6699:e29:b0:6c:f434:64f4]:4001",
    "status": "INVALID_ARGUMENT"
  }
}
```
<img width="400" alt="image" src="https://github.com/GoogleCloudPlatform/magic-modules/assets/10195482/cbcb1355-8eb9-4281-a5ef-dbcbd2018a41">

So yes it seems to be required!

Reply:
Thanks for looking into this! Sometimes the API documentation explicitly says a field is required and sometimes it doesn't (when it should) so I figured I'd ask